### ISSUE TITLE
segmentation violation while "devspace purge"

### ISSUE THREAD

**Issue by leventogut** [loft-sh/devspace#1361]
**What happened?**  
segmentation violation

**What did you expect to happen instead?**  
purge

**How can we reproduce the bug?** (as minimally and precisely as possible)  
it doesn't happen all the time, need to analyze the relevant code to see if there is a bug we can find.

**Local Environment:**  
- DevSpace Version: devspace version 5.9.1
- Operating System: mac
- Deployment method: helm 

**Kubernetes Cluster:**  
- Cloud Provider: other / docker0-desktop
- Kubernetes Version:
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:15:20Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"

**Anything else we need to know?**  
```
rasmusX:php-laravel-nginx-devspace logut$ devspace purge
[info]   Using namespace 'default'
[info]   Using kube context 'docker-desktop'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x52e074a]

goroutine 1 [running]:
github.com/loft-sh/devspace/pkg/devspace/hook.(*executer).Execute(0xc000010e28, 0x58dd9ce, 0x5, 0x59753b0, 0x10, 0xc0010aab58, 0x5, 0x0, 0x0, 0x5d05d78, ...)
        /Users/runner/work/devspace/devspace/pkg/devspace/hook/hook.go:138 +0x150a
github.com/loft-sh/devspace/pkg/devspace/deploy.(*controller).Purge(0xc0012bc7e0, 0x0, 0x0, 0x0, 0x5d07140, 0x6fdd0a0, 0xc0012bc7e0, 0x58e0c8c)
        /Users/runner/work/devspace/devspace/pkg/devspace/deploy/deploy.go:282 +0x984
github.com/loft-sh/devspace/cmd.(*PurgeCmd).Run(0xc0000b7b30, 0x5d03508, 0x7035488, 0x7035488, 0x0, 0x0, 0xc0010738c0, 0x7035488, 0x0, 0x0, ...)
        /Users/runner/work/devspace/devspace/cmd/purge.go:129 +0x6f3
github.com/loft-sh/devspace/cmd.NewPurgeCmd.func1(0xc0010738c0, 0x7035488, 0x0, 0x0, 0x0, 0x0)
        /Users/runner/work/devspace/devspace/cmd/purge.go:54 +0x89
github.com/spf13/cobra.(*Command).execute(0xc0010738c0, 0x7035488, 0x0, 0x0, 0xc0010738c0, 0x7035488)
        /Users/runner/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:842 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc000392580, 0x7035488, 0xc000392580, 0x0)
        /Users/runner/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/runner/work/devspace/devspace/vendor/github.com/spf13/cobra/command.go:887
github.com/loft-sh/devspace/cmd.Execute()
        /Users/runner/work/devspace/devspace/cmd/root.go:111 +0x7d
main.main()
        /Users/runner/work/devspace/devspace/main.go:20 +0xf7
rasmusX:php-laravel-nginx-devspace logut$ 
```

```
version: v1beta9
images:
  php:
    image: ${PHP_IMAGE}
    preferSyncOverRebuild: true
    injectRestartHelper: true
    appendDockerfileInstructions:
    - USER root
    dockerfile: Dockerfile
deployments:
- name: ${APP_DEPLOYMENT_NAME}
  helm:
    componentChart: true
    values:
      containers:
      - name: nginx
        image: ${NGINX_IMAGE}
        volumeMounts:
        - containerPath: /etc/nginx/nginx.conf
          volume:
            name: ${NGINX_CONFIG_CONFIGMAP_NAME}
            subPath: nginx.conf
            readOnly: true
        - containerPath: /shared
          volume:
            name: ${SHARED_VOLUME_NAME}
            subPath: /shared
            readOnly: true
        command:
        - "/bin/sh"
        args:
        - "-c"
        - "mkdir -p /var/www/html/public && touch /var/www/html/public/index.php && nginx -g 'daemon off;'"
      - name: php
        image: ${PHP_IMAGE}
        command:
        - "/bin/sh"
        args:
        - "-c"
        - "npm run dev && php /var/www/html/artisan migrate && php-fpm"
        volumeMounts:
        - containerPath: /shared
          volume:
            name: ${SHARED_VOLUME_NAME}
            subPath: /shared
            readOnly: false
      volumes:
      - name: nginx-config
        configMap:
          name: nginx-config
      - name: ${SHARED_VOLUME_NAME}
        size: ${SHARED_VOLUME_SIZE}
      labels:
        app: nginx
        portforward: true
- name: ${MYSQL_DEPLOYMENT_NAME}
  helm:
    componentChart: false
    chart:
      name: ${MYSQL_CHART}
      version: ${MYSQL_CHART_VERSION}
    values:
      auth:
        rootPassword: ${MYSQL_ROOT_PASSWORD}
        database: ${MYSQL_DATABASE}
        username: ${MYSQL_USER}
        password: ${MYSQL_PASSWORD}
- name: ${REDIS_DEPLOYMENT_NAME}
  helm:
    componentChart: false
    chart:
      name: ${REDIS_CHART}
      version: ${REDIS_CHART_VERSION}
    values:
      networkPolicy:
        enabled: true
        allowExternal: true
      password: ${REDIS_PASSWORD}
      cluster:
        enabled: false
dev:
  ports:
  - labelSelector:
      app: nginx
      portforward: true
    forward:
    - port: 8080
      remotePort: 80
  open:
  - url: http://localhost:8080
  sync:
  - imageName: php
    excludePaths:
    - .git/
    uploadExcludePaths:
    - Dockerfile
    - "*.Dockerfile"
    - devspace.yaml
    - vendor/*
    - node_modules/*
    onUpload:
      restartContainer: true
profiles:
- name: production
  patches:
  - op: remove
    path: images.app.appendDockerfileInstructions
  - op: remove
    path: images.app.injectRestartHelper
- name: interactive
  patches:
  - op: add
    path: dev.interactive
    value:
      defaultEnabled: true
  - op: add
    path: images.app.entrypoint
    value:
    - sleep
    - "9999999999"
commands:
- name: php
  command: devspace enter php
vars:
hooks:
- command: "/bin/sh"
  args:
    - "-c"
    - "kubectl delete configmap ${NGINX_CONFIG_CONFIGMAP_NAME} || true; kubectl create configmap ${NGINX_CONFIG_CONFIGMAP_NAME} --from-file=./deploy/nginx.conf"
  when:
    before:
      deployments: ${APP_DEPLOYMENT_NAME}
- command: "/bin/sh"
  args:
    - "-c"
    - "kubectl delete configmap ${NGINX_CONFIG_CONFIGMAP_NAME}"
  when:   
    after:
      purgeDeployments: ${APP_DEPLOYMENT_NAME}
```

/kind bug

**Comment by leventogut** [comment]
Code is here: (and I just had the same segmentation fault over and over so it is reproducible, I was mistaken:

https://github.com/leventogut/php-laravel-nginx-devspace/tree/index.php

**Comment by FabianKramm** [comment]
@leventogut thanks a lot for opening this issue! We'll include a fix for this in the next version!

**Comment by FabianKramm** [comment]
@leventogut should be fixed with v5.10.0
